lib/collection/src/shards/local_shard.rs

```diff
@@ -395,9 +395,20 @@ impl LocalShard {
         bar.set_style(progress_style);
 
         bar.set_message(format!("Recovering collection {collection_id}"));
+
         let segments = self.segments();
-        // ToDo: Start from minimal applied version
-        for (op_num, update) in wal.read_all() {
+
+        // TODO: Should we use *most* or *least* recent version here? 🤔
```

Comment:
I think we want the least recent (oldest) version. 

Otherwise older segments could be missing data when replaying.

@generall should confirm this though.

Reply:
Yeah, I'm not 100% sure about it as well. I guess we can also imagine a situation when we have, say, an empty segment (or the one with a version ~1000 operations older than all other segments), so the least recent version would be 0, and we'd re-apply "outdated" operations. I'm not 100% sure what our guarantees are regarding segment version.

Reply:
> and we'd re-apply "outdated" operations. I'm not 100% sure what our guarantees are regarding segment version.

I don't think that causes data integrity issues. We'd better replay too much than too little.

IIRC we don't have operations that would break when the segment becomes too old.